src/functions.php

```diff
@@ -478,23 +478,48 @@ function is_settled(PromiseInterface $promise)
 function coroutine(callable $generatorFn)
 {
     $generator = $generatorFn();
-    return __next_coroutine($generator->current(), $generator)->then();
+    $state = new \stdClass;
```

Comment:
> Additionally, it would be easier to test in isolation
What would be the benefit in testing such a trivial piece of code in isolation? 
> which is important in a project that accepts pull requests from the community
In this case, why would it be valuable?
> I would need to see a performance test to gauge which approach is faster.
Additional function calls make things slower. It doesn't really require testing.

Given that this is a trivial change, non-breaking change which fixes a pretty serious bug it's pretty frustrating to be debating design decisions which provide no tangible benefits but require extra effort and would slow things down. I'm closing this issue as this is a waste of my time. Please fix this bug.